lib/exsoda/configuration_reader.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Exsoda.ConfigurationReader do
+  alias Exsoda.Http
+  alias Exsoda.Configuration
+
+  defmodule GetConfig do
+    defstruct type: nil,
+      default_only: nil,
```

Comment:
`default_only: false` will default it to `true` or `false`, then you don't need the get_config_defaults